model/auth.go

```diff
@@ -232,6 +241,10 @@ type OAuth2AuthProperties struct {
 	RequestedIssuer string `json:"requestedIssuer,omitempty" validate:"omitempty,min=1"`
 }
 
+func (b *OAuth2AuthProperties) DeepCopyAuthProperties() AuthProperties {
```

Comment:
Is this function auto generated？

Reply:
I don't think so. It looks like a workaround. @spolti, can you explain, please? Maybe also add a comment on these methods.

Reply:
No, it isn't auto generated, deepcopy requires this method on interfaces.

Reply:
added a comment in the interface definition.